virttest/utils_test/__init__.py

```diff
@@ -1969,19 +1976,17 @@ def _git_download(self, url, destination):
         destination = os.path.join(destination, self.base_name)
         git.get_repo(url, destination_dir=destination)
 
-    def _tarball_download(self, url, destination):
+    def _file_download(self, url, destination):
         """
-        tarball download and uncompress
-        :param url: dowload url
+        download and uncompress tar / XZ file
+        :param url: download url
         :param destination: download destination
         """
-        import tarfile
-        tarball_path = os.path.join(destination, self.base_name)
-        download.get_file(url, tarball_path)
-        tarball = tarfile.open(tarball_path)
-        tarball.extractall(destination)
-        self.base_name = tarball.members[0].name.split("/")[0]
-        tarball.close()
+        file_path = os.path.join(destination, self.base_name)
+        download.get_file(url, file_path)
+        file_type = process.getoutput("file %s" % file_path)
+        if "directory" not in file_type:
+            self.base_name = archive.uncompress(file_path, destination)
```

Comment:
@balamuruhans , archive.uncompress returns "None" here, so self.base_name=None

Reply:
@aliang123 , I think you have missed to include Avocado patch - https://github.com/avocado-framework/avocado/pull/2705, that returns the extracted path.

Reply:
@balamuruhans, test with patch avocado-framework/avocado#2705, self.base_name still get a wrong value when i run iozone test...
current:
    self.base_name = iozone3_458/src/current/Changes.txt
correct:
    self.base_name= iozone3_458

So, with patch #2750, we can't get a correct self.base_name

Reply:
@aliang123 yes, correct I have tested it for stress-ng but not for iozone and `tarfile` module return the `getnames()` differently for iozone and stress-ng.
I have fixed in PR https://github.com/avocado-framework/avocado/pull/2743.
Thank you for reporting it and making code better.

Reply:
@balamuruhans, yes, with patch #2743, download-iozone can work normally.
So, after #2743 merged, i'll give my ack on this patch.

Reply:
yep :+1: I would wait for your ACK

Reply:
@aliang123, from avocado `archive.uncompress()` will return the first member of the extracted tar/zip file, the users have to handle it accordingly. so please see L1992 for it.